### PR TITLE
Update FASTQ file merging to use sample relations

### PR DESCRIPTION
**Pull request by AnzeLovse**
## Checklist

* [x] Update CHANGELOG.rst for each commit separately:
  * Pay attention to write entries under the "Unreleased" section.
  * Mark all breaking changes as "**BACKWARD INCOMPATIBLE:**" and put them
    before non-breaking changes.
  * If a commit modifies a feature listed under "Unreleased" section,
    it might be sufficient to modify the existing CHANGELOG entry from previous
    commit(s).
* [x] Bump the process version:
  * **MAJOR version (first number)**: Backward incompatible changes (changes
    that break the api/interface). Examples: renaming the input/output, adding
    mandatory input, removing input/output...
  * **MINOR version (middle number)**: add functionality or changes in a
    backwards-compatible manner. Examples: add output field, add non-mandatory
    input parameter, use a different tool that produces same results...
  * **PATCH version (last number)**: changes/bug fixes that do not affect
    the api/interface. Examples: typo fix, change/add warning messages...
* [x] All inputs are used in process.
* [x] All output fields have their ``re-save`` calls.

### CODE REVIEW

**Comment by AnzeLovse** [comment]
@jkokosar could you please review this PR?

File: resolwe_bio/processes/support_processors/merge_samples.py [hunk 1]
```diff
@@ -1,218 +1,190 @@
...
+
+            self.run_process(
+                slug="upload-fastq-single",
+                inputs={
+                    "src": symlinks,
+                    "merge_lanes": True,
+                },
+            )
+
+            mereged_objects = Data.filter(entity__name=symlinks[0])
```
Comment (jkokosar) [inline resolwe_bio/processes/support_processors/merge_samples.py:111]: `mereged` -> `merged` here and below.

File: resolwe_bio/processes/support_processors/merge_samples.py [hunk 2]
```diff
@@ -1,218 +1,190 @@
...
-
-        outputs.fastqc_url = fastqc_url
-        outputs.fastqc_archive = [fastqc]
+
+        # Check if user has selected multiple read objects from the same sample
+        data_by_sample = {}
+        for data in inputs.reads:
+            if data.entity_id in data_by_sample:
+                self.warning(
+                    "There are multiple read objects for "
```
Comment (jkokosar) [inline resolwe_bio/processes/support_processors/merge_samples.py:85]: It is common for the sample to have trimmed reads objects attached. For any subsequent joining of already analysed data, the original input (FASTQ) that originated the entity might make more sense. So, I propose to merge the first FASTQ on the sample, not the last.

File: docs/CHANGELOG.rst [hunk 3]
```diff
@@ -22,6 +22,9 @@ Added
 
 Changed
 -------
+- **BACKWARD INCOMPATIBLE:** Update ``merge-fastq-single`` and
```
Comment (jkokosar) [inline docs/CHANGELOG.rst:25]: This should go to the unreleased section

File: resolwe_bio/processes/support_processors/merge_samples.py (hunk 2 above)
Comment (AnzeLovse) [reply resolwe_bio/processes/support_processors/merge_samples.py:85]: Thanks that is a good suggestion.

**Pull request merged**